storage/filesystem/object.go

```diff
@@ -133,6 +135,23 @@ func (s *ObjectStorage) EncodedObject(t plumbing.ObjectType, h plumbing.Hash) (p
 		obj, err = s.getFromPackfile(h, false)
 	}
 
+	// If the error is still object not found, check if it's a shared object
+	// repository.
+	if err == plumbing.ErrObjectNotFound {
+		alt, e := s.dir.Alternates()
+		if e == nil {
+			// Create a new object storage pointing to the path in the
+			// alternates file.
+			fs := osfs.New(filepath.Dir(alt))
+			dir := dotgit.New(fs)
+			o, oe := newObjectStorage(dir)
```

Comment:
Maybe this secondary storage object should be cached rather than recreated each time we decode an object stored in the alternates?